### ISSUE TITLE
[Feature] SCAN limit and return cursor

### ISSUE THREAD

**Issue by fa-sharp** [aembke/fred.rs#313]
**Describe the solution you'd like**
Actually not sure if this is a feature or bug - I would like an ergonomic way to stop the SCAN streams when they hit a certain number of results, and then return the last cursor (or just `"0"` if it got all the results). Then the SCAN commands would also have an `Option<String>` parameter to provide the cursor to start from. Basically, this would allow us to "paginate" the results.

Something like: `client.scan(pattern, count, Some(cursor), Some(max_count))`

**Additional context**
The way I'm doing this right now does work, but it feels clunky, and I see errors in the DEBUG logs that indicate that the client is still scanning all the results after I break the loop. Maybe there's a better way to do this, so do let me know if so.

Current code (abbreviated):
```rust
let count: u32 = 100;
let max_count: usize = 1000;
let mut scan_stream = client.scan("foo:*", Some(count), None);
let mut keys: Vec<String> = vec![];
let mut current_cursor: Option<String> = None;

while let Some(mut scan_result) = scan_stream.try_next().await? {
    current_cursor = scan_result.cursor().map(|c| c.to_string());
    if let Some(page) = scan_result.take_results() {
        keys.extend(
            page.iter()
                .map(|key| key.as_str_lossy()),
        );
    }
    let _ = scan_result.next();
    if keys.len() >= max_count {
        break;
    }
}
```

When there are more results than `max_count` and the loop breaks early, here are the DEBUG errors I see. I assume this means that the client is still scanning all pages:
`fred-RB7FokOcij: Error sending SCAN page.`
`fred-RB7FokOcij: Error sending SCAN page.`
`fred-RB7FokOcij: Error sending SCAN page.`
....

**Comment by aembke** [comment]
Hey @fa-sharp, this'll be added in the next version (https://github.com/aembke/fred.rs/pull/314). Here's an example of what the interface will look like: https://github.com/aembke/fred.rs/blob/feat/10.0.0/examples/scan.rs#L52-L74

**Comment by fa-sharp** [comment]
Looks great, thank you!! I ended up using a custom command to do something similar in the meantime, but that should work nicely once it's released.

**Comment by aembke** [comment]
Sounds good, I just merged v10.0.0 so I'll close this for now.
